extensions/amp-story/1.0/amp-story-reaction.js

```diff
@@ -225,47 +257,30 @@ export class AmpStoryReaction extends AMP.BaseElement {
       true /** callToInitialize */
     );
 
-    // Add a click listener to the element to trigger the class change
-    this.rootEl_.addEventListener('click', (e) => this.handleTap_(e));
-  }
-
-  /**
-   * Handles a tap event on the quiz element.
-   * @param {Event} e
-   * @private
-   */
-  handleTap_(e) {
-    if (this.hasUserSelection_) {
-      return;
-    }
-
-    const optionEl = closest(
-      dev().assertElement(e.target),
-      (element) => {
-        return element.classList.contains('i-amphtml-story-reaction-option');
-      },
-      this.rootEl_
+    this.optionElements_ = toArray(
+      this.rootEl_.querySelectorAll('.i-amphtml-story-reaction-option')
     );
 
-    if (optionEl) {
-      this.handleOptionSelection_(optionEl);
-    }
+    this.optionElements_.forEach((element, reactionValue) => {
+      element.addEventListener('click', (unusedEvent) =>
```

Comment:
With just one listener, the click listener had to search up the DOM on every click to find an option and then find the index of the option. Besides, making a list of the options on the abstract class was useful on the quiz implementation for updating the option percentages. Also the index was retrieved from a property on each option node that was created on the quiz component, which seemed like delegating that responsibility to all the implementations (instead of keeping track of the option index on the abstract side) didn't look right. That's why adding a listener that already had option index made sense to avoid all those traversals and responsibilities from the concrete classes.

I can change it back, but I'm curious, why is the one-listener approach more convenient in this case?